lib/elixir/test/elixir/code_normalizer/quoted_ast_test.exs

```diff
@@ -629,6 +629,43 @@ defmodule Code.Normalizer.QuotedASTTest do
       assert quoted_to_string(quote(do: foo |> [bar: :baz])) == "foo |> [bar: :baz]"
     end
 
+    test "keyword arg edge case: cursor" do
+      input = "def foo, do: :bar, __cursor__()"
+      expected = "def foo, [{:do, :bar}, __cursor__()]"
+
+      ast = Code.string_to_quoted!(input)
+
+      assert quoted_to_string(ast) == expected
+
+      ast =
+        Code.string_to_quoted!(input,
+          literal_encoder: &{:ok, {:__block__, &2, [&1]}}
+        )
+
+      assert quoted_to_string(ast) == expected
+    end
+
+    test "keyword arg edge case: literal encoder" do
```

Comment:
This test is a reverse engineering attempt to understand how the `match?` clause works, but it doesn't seem to